### DISCUSSION TITLE
Why text or markup is allowed after an inlined return?

### DISCUSSION THREAD

**Discussion by uben0** [typst/typst#2103] [General]
This isn't allowed:
```typst
#let c = a + b Hello World
```
But this is valid:
```typst
#return a + b Hello World
```
I am writing a TreeSitter grammar, and this feature of Typst bugs me a lot. It feels unintentional and inconsistent. Should I report this as an issue?

**Comment by Enivex** [comment]
You say it's valid, but neglect to say what it actually does

**Reply by uben0** [reply]
What do you mean?

**Reply by uben0** [reply]
```typst
#let a() = {
  [#return 3 + 5 Hello World]
}

#a()
```
This will render `8`.

**Reply by laurmaedje** [reply]
You can open an issue.

**Reply by uben0** [reply]
https://github.com/typst/typst/issues/2104
